### DISCUSSION TITLE
Cannot execute code. Unable to start a runtime for the R language.

### DISCUSSION THREAD

**Discussion by petemeng** [posit-dev/positron#4288] [Q&A]
I am experiencing problems running R code in Positron. When I try to execute R code (e.g., library(tidyverse)), I receive the following error:
```
Cannot execute code. Unable to start a runtime for the R language.
```
I attempted to install the R extension, but received a message indicating that it conflicts with Positron's built-in features:

**Comment by DavisVaughan** [comment]
Positron already comes bundled with full support for R, so please don't install the R extension that you may have used in VS Code.

If you uninstall it, do you get the same issues?

**Comment by petemeng** [comment]
I'm not going to install the R extension anymore, but I'm still encountering the issue: "Cannot execute code. Unable to start a runtime for the R language." It seems like Positron isn't detecting R. Is there a way to resolve this problem?

![image](https://github.com/user-attachments/assets/fb822283-9715-4bf2-8532-23b292c9d75c)

**Comment by DavisVaughan** [comment]
Can you please switch to the `OUTPUT` tab, click on the drop down and change to `Positron R Extension`, and copy and paste in everything you see there?

**Comment by jennybc** [comment]
Probably converging with #4280.

@petemeng Were you previously using Positron successfully, with R?

**Reply by petemeng** [reply]
No, this is my first to use Positron.

**Comment by petemeng** [comment]
```
2024-08-08 23:38:24.657 [info] Log level: Debug
2024-08-08 23:38:24.657 [info] Refreshing testthat status
2024-08-08 23:38:24.657 [info] Not working in an R package
2024-08-08 23:38:24.657 [info] Context key 'testthatIsConfigured' is 'false'
2024-08-08 23:38:24.657 [info] Context key 'testthatHasTests' is 'false'

```

**Comment by petemeng** [comment]
![image](https://github.com/user-attachments/assets/8d01e0a3-fcfa-4c91-b434-3b1e12bb6fcb)

I am sure that i have installed R in my system!

**Comment by ZhangMH2000** [comment]
Hi Positron developers. I'm using Ubuntu 22.04, and I downloaded Positron [2024.08.0-48]. When I attempted to run R scripts, it outputs the error "Cannot execute code. Unable to start a runtime for the R language." I'm pretty sure R 4.3.2 has been installed in this conda environment. I looked #4280  and I found that [2024.08.0-31] version addresses the issue impacting  Positron Windows installations locating versions of R. The deb version of [2024.08.0-31] also showed the same error. Do you have any suggestions? Thank you!

**Comment by DavisVaughan** [comment]
@ZhangMH2000 can you please open a new discussion thread for your issue? It's easier to track it on its own as you probably have a separate unique issue
